src/payload_size.rs

```diff
@@ -23,10 +23,11 @@
 pub const HEADER_SIZE_BYTES: u64 = 8;
 
 #[allow(dead_code)] // TODO No need for this func anymore?
-/// How many chunks of `chunk_size_bytes` to send for a given payload of `payload_size_bytes`
-/// Taking into account the overhead of HEADER_SIZE_BYTES per chunk
+/// How many chunks of `chunk_size_bytes` to send for a given payload of
+/// `payload_size_bytes` Taking into account the overhead of `HEADER_SIZE_BYTES`
+/// per chunk
 pub fn number_chunks_overhead(payload_size_bytes: u64, chunk_size_bytes: u16) -> u64 {
-    let chunk_payload_size_bytes: u64 = (chunk_size_bytes as u64) - HEADER_SIZE_BYTES;
+    let chunk_payload_size_bytes: u64 = (u64::from(chunk_size_bytes)) - HEADER_SIZE_BYTES;
```

Comment:
I have a problem with chunk size / total file size types right now, that goes beyond this, and I appreciate your input on it:

- I know that chunk IDs are limited (by my unwritten "file format spec") to 3 digits  ==>  999 chunks max (because who will scan 1001 QR codes?). T hat means using u16 for smallest size that's appropriate to hold the data.
- For file size in bytes of file to exfil, theoretical max payload is ~10MB (if I up chunk size above 1KB and stay <1000 chunks) which is  > 65536, so it must be at least u32 (I used u64, whatevs).
- Reading bytes off a file usually asks for `usize` which is neither here nor there.
- Math operations between types is annoying (substracting usize from u64 etc, dividing to find out how many chunks...)

What do you think? I'm close to considering everything to be either u64 or usize, whichever works, because I had strong opinions initially but I really do not care that much

Reply:
yeah i dunno. Rust has pretty strong opinions about the ways you're allowed to mix operations between primitive types, as you can see.

This particular edit is actually an allowed, non-lossy transformation. by using the `from` syntax instead of the casting syntax, you'll get a compiler error if you ever change the data type into something that doesn't have a non-lossy transformation.